daml/Marketplace/Custody/Model.daml

```diff
@@ -145,3 +164,11 @@ createNewVersionOfDescription description postAssetVersion serializedRemainingCl
     claims = serializedRemainingClaims
  where
   adjustedId = description.assetId with version = postAssetVersion
+
+-- TODO this is still not unique e.g. if I have two asset deposits with the same asset
+createTradeId : Id -> Int -> Id
+createTradeId assetId transferNumber =
+  Id with
+    signatories = assetId.signatories
+    label = assetId.label <> " " <> show transferNumber <> "th transfer"
```

Comment:
Doesn't this also break when you do e.g. settlement of two consecutive bond coupons, as you're not using the version anywhere?

Reply:
No, I added such a test.
Note that in such a case the 'transferNumber' is different between transactions of the same lifecycle (e.g. in the last coupon payment where there are two tx-es: 1. payout for the coupon, 2. payout for the notional) and version is different between transactions of different lifecycles (e.g. payment of 1st and 2nd coupon).